interpreter.py

```diff
@@ -0,0 +1,164 @@
+                        printf_args.append(self.global_state.get_variable(rffi.cast(rffi.INT, arg)))
+                    elif LLVMIsConstant(arg):
+                        printf_args.append(LLVMConstIntGetSExtValue(arg))
+                    else:
+                        # raise custom exception
+                        print "error"
+                print string_format % tuple(printf_args)
+            else:
+                raise NotImplementedError(fn_name)
+        elif opcode == LLVMAlloca:
+            pass
+        elif opcode == LLVMStore:
+            pass
+
+    def run(self, function):
+        frame = State()
+
+        block = LLVMGetFirstBasicBlock(function)
+        while block:
+            instruction = LLVMGetFirstInstruction(block)
+            while instruction:
+                opcode = LLVMGetInstructionOpcode(instruction)
+
+                operand_list = [LLVMGetOperand(instruction, i) \
+                                for i in range(0,  LLVMGetNumOperands(instruction))]
+
+                value = self.exec_operation(frame, opcode, operand_list)
+                frame.set_variable(rffi.cast(rffi.INT, instruction), value)
+
+                instruction = LLVMGetNextInstruction(instruction)
+
+            block = LLVMGetNextBasicBlock(block)
+        print "Done.\n[INFO]", frame.vars, frame.var_offsets.items()
+
+def main(args):
+    if len(args) < 2:
+        print"[ERROR]: Need an argument:\nUsage: ./llvmtest name.bc [C args]\n"
+        return 1
+
+    main_argc = len(args) - 2
+    main_argv = args[2:]
+
+    module = LLVMModuleCreateWithName("module_test")
+
+    with lltype.scoped_alloc(rffi.CCHARPP.TO, 1) as out_message:
+        mem_buff = lltype.malloc(rffi.VOIDP.TO, 1, flavor="raw")
+        with lltype.scoped_alloc(rffi.VOIDPP.TO, 1) as mem_buff_ptr:
+            mem_buff_ptr[0] = mem_buff
+            rc = LLVMCreateMemoryBufferWithContentsOfFile(args[1], mem_buff_ptr, out_message)
+            if rc != 0:
+                print"[ERROR]: Cannot create memory buffer with contents of"\
+                     " %s: %s.\n" % (args[1], rffi.charp2str(out_message[0]))
+                return 2
+            mem_buff = mem_buff_ptr[0]
+
+        with lltype.scoped_alloc(rffi.VOIDPP.TO, 1) as module_ptr:
+            module_ptr[0] = module
+            rc = LLVMParseBitcode(mem_buff, module_ptr, out_message)
+            if rc != 0:
+                print "[ERROR]: Cannot parse %s: %s.\n" % (args[1], rffi.charp2str(out_message[0]))
+                return 3
+            module = module_ptr[0]
+        main_fun = LLVMGetNamedFunction(module, "main")
+
+    with lltype.scoped_alloc(rffi.VOIDPP.TO, 1) as basic_blocks_main_ptr:
+        LLVMGetBasicBlocks(main_fun, basic_blocks_main_ptr)
+        basic_blocks_main = basic_blocks_main_ptr[0]
+
+    global_state = State()
+    global_var = LLVMGetFirstGlobal(module)
+
+    # assuming all global_vars are strings
```

Comment:
Is this a safe assumption? Should be have an assertion or exit with a message if we find a non-string?